cmd/subnetcmd/removeValidator.go

```diff
@@ -0,0 +1,237 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/subnet"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanchego/genesis"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/spf13/cobra"
+)
+
+// avalanche subnet deploy
+func newRemoveValidatorCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "removeValidator [subnetName]",
+		Short: "Remove a permissioned validator from your subnet",
+		Long: `The subnet removeValidator command stops a whitelisted, subnet network validator from
+validating your deployed Subnet.
+
+To remove the validator from the Subnet's allow list, you first need to provide
+the subnetName and the validator's unique NodeID. You can bypass
+these prompts by providing the values with flags.`,
+		SilenceUsage: true,
+		RunE:         removeValidator,
+		Args:         cobra.ExactArgs(1),
+	}
+	cmd.Flags().StringVarP(&keyName, "key", "k", "", "select the key to use [fuji deploy only]")
+	cmd.Flags().StringVar(&nodeIDStr, "nodeID", "", "set the NodeID of the validator to add")
+	cmd.Flags().BoolVar(&deployLocal, "local", false, "remove from the locally deployed Subnet")
+	cmd.Flags().BoolVar(&deployTestnet, "fuji", false, "remove from `fuji` deployment (alias for `testnet`)")
+	cmd.Flags().BoolVar(&deployTestnet, "testnet", false, "remove from `testnet` deployment (alias for `fuji`)")
+	cmd.Flags().BoolVar(&deployMainnet, "mainnet", false, "remove from `mainnet` deployment")
+	cmd.Flags().StringSliceVar(&subnetAuthKeys, "subnet-auth-keys", nil, "control keys that will be used to authenticate the removeValidator tx")
+	cmd.Flags().StringVar(&outputTxPath, "output-tx-path", "", "file path of the removeValidator tx")
+	cmd.Flags().BoolVarP(&useLedger, "ledger", "g", false, "use ledger instead of key (always true on mainnet, defaults to false on fuji)")
+	cmd.Flags().StringSliceVar(&ledgerAddresses, "ledger-addrs", []string{}, "use the given ledger addresses")
+	return cmd
+}
+
+func removeValidator(_ *cobra.Command, args []string) error {
+	var (
+		nodeID ids.NodeID
+		err    error
+	)
+
+	var network models.Network
+	switch {
+	case deployTestnet:
+		network = models.Fuji
+	case deployMainnet:
+		network = models.Mainnet
+	}
+
+	if network == models.Undefined {
+		networkStr, err := app.Prompt.CaptureList(
+			"Choose a network to remove a validator from",
+			[]string{models.Local.String(), models.Fuji.String(), models.Mainnet.String()},
```

Comment:
should we enable addValidator to local also?

Reply:
we could, but that would be a little more involved since we'd also need to get the network runner to start one.